### PR TITLE
[#4095] Migrate Microsoft.Bot.Builder.Dialogs.Declarative.Tests project to xUnit

### LINKED ISSUE #4095: migrate builder.diaglogs.declarative to xunit
#4039

### PR DESCRIPTION
**Pull request by denscollo**
Fixes #4095

## Description
This PR migrates all [Microsoft.Bot.Builder.Dialogs.Declarative.Tests](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests) tests from **MSTest** to **xUnit**.

## Detailed Changes
- Removed MSTest packages from the .csproj.
- Replaced `[TestMethod]` with `[Fact]`.
- Changed Asserts.
- Replaced `Assert.Fail` with `thown new XunitException`.
- Removed `TestContext`.
- Replaced `ClassInitialize` method with a class' constructor to execute tests' shared code.
- Used constants instead of variables where applicable.
- Used `var` instead of the specific type.
- Removed redundant parenthesis.

Also, we did the following changes in: 
- **_JsonLoadTests_**:
    - Added the `Dispose` method to clean up after each test execution.
    - Added the test name as parameter in `BuildTestFlow` and `BuildQnAMakerTestFlow` to pass it to `InitializeAdapter` method instead of `TestContext.TestName`
- **_SchemaMergeTests_**:
   - Replaced `[DataTestMethod]` with `Theory`
- **_Startup_**:
  - Replaced `[AssemblyInitialize]` with the `AssemblyFixture` feature of Xunit. 

## Testing
The following image shows the tests passing after the changes.
image

### CODE REVIEW

**Review by EricDahlvang — CHANGES_REQUESTED** [review]
:clock1:

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs [hunk 1]
```diff
@@ -1,56 +1,42 @@
...
+using Xunit;
 
 namespace Microsoft.Bot.Builder.Dialogs.Loader.Tests
 {
-    [TestClass]
-    public class JsonLoadTests
+    public class JsonLoadTests : IClassFixture<JsonLoadFixture>
     {
-        private static ResourceExplorer resourceExplorer;
+        // private static ResourceExplorer _resourceExplorer;
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadTests.cs]: Feedback applied. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadFixture.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.IO;
+using Microsoft.Bot.Builder.Dialogs.Declarative.Resources;
+
+namespace Microsoft.Bot.Builder.Dialogs.Loader.Tests
+{
+    public class JsonLoadFixture : IDisposable
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/JsonLoadFixture.cs]: We renamed and updated the class implementation to be similar to the one you mentioned. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/Startup.cs [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
+        public Startup(IMessageSink messageSink)
+            : base(messageSink)
         {
             ComponentRegistration.Add(new DeclarativeComponentRegistration());
             ComponentRegistration.Add(new AdaptiveComponentRegistration());
             ComponentRegistration.Add(new LanguageGenerationComponentRegistration());
             ComponentRegistration.Add(new QnAMakerComponentRegistration());
         }
+
+        public new void Dispose()
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/Startup.cs]: Done. Thanks!

File: tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceFixture.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Bot.Builder.Dialogs.Declarative.Tests
+{
+    public class ResourceFixture : IDisposable
```
[the remark this answers is not in the document]
Comment (ceciliaavila) [reply tests/Microsoft.Bot.Builder.Dialogs.Declarative.Tests/ResourceFixture.cs]: We renamed the class.
The deletion of the files was in the _ClassInitialize_, that’s why we included it in the fixture, so it runs only once, before all tests.

**Review by EricDahlvang — APPROVED** [review]

**Pull request merged**